core-java-modules/core-java-14/src/test/java/com/baeldung/java14/recordsCustomConstructors/StudentRecordJUnitTest.java

```diff
@@ -20,18 +26,46 @@ public void givenStudentRecordData_whenCreated_thenStudentPropertiesMatch() {
         assertEquals(2, s2.rollNo());
         assertEquals(80, s2.marks());
     }
+
     @Test
-    public void givenStudentRecordsList_whenSortingDataWithName_thenStudentsSorted(){
-        List<StudentRecord> studentRecords = List.of(
-                new StudentRecord("Dana", 1, 85),
-                new StudentRecord("Jim", 2, 90),
-                new StudentRecord("Jane", 3, 80)
-        );
+    void givenStudentRecordsList_whenSortingDataWithName_thenStudentsSorted() {
+        List<StudentRecord> studentRecords = List.of(new StudentRecord("Dana", 1, 85), new StudentRecord("Jim", 2, 90), new StudentRecord("Jane", 3, 80));
 
         List<StudentRecord> mutableStudentRecords = new ArrayList<>(studentRecords);
         mutableStudentRecords.sort(Comparator.comparing(StudentRecord::name));
 
         List<StudentRecord> sortedStudentRecords = List.copyOf(mutableStudentRecords);
-        assertEquals("Jane", sortedStudentRecords.get(1).name());
+        assertEquals("Jane", sortedStudentRecords.get(1)
+            .name());
+    }
+
+    @Test
+    void whenCreateStudentRecordWithInvalidMarks_thenRaiseException() {
+        assertThrows(IllegalArgumentException.class, () -> new StudentRecord("Jane", 2, 150));
+    }
+
+    @Test
+    void whenCreateStudentRecordV2_thenCorrect() {
+        StudentRecordV2 studentV2 = new StudentRecordV2("Jane", 2, 85);
+        assertEquals('B', studentV2.grade());
+    }
+
+    @Test
+    void whenCreateStudentRecordV3_thenCorrect() {
+        StudentRecordV3 studentV3 = new StudentRecordV3("Jane");
+
+        assertFalse(studentV3.id().isEmpty());
+        assertTrue(studentV3.hobbies().isEmpty());
```

Comment:
I think we have the guideline to use AssertJ for complex assertions (like checking a collection is not empty) rather than using assertTrue/False with boolean conditions

Reply:
I converted them to assertj assertions.